lib/datasource/gitlab-releases/index.ts

```diff
@@ -0,0 +1,46 @@
+import { cache } from '../../util/cache/package/decorator';
+import { Datasource } from '../datasource';
+import type { GetReleasesConfig, Release, ReleaseResult } from '../types';
+import type { GitlabRelease } from './types';
+
+export class GitlabReleasesDatasource extends Datasource {
+  static readonly id = 'gitlab-releases';
+
+  readonly defaultRegistryUrls = ['https://gitlab.com'];
+
+  static readonly registryStrategy = 'first';
+
+  constructor() {
+    super(GitlabReleasesDatasource.id);
```

Comment:
I think we also need to override `Http` by `GitlabHttp` class, so we reuse buildin paging.